App.js

```diff
@@ -256,6 +257,13 @@ export default class App extends React.Component {
     this.state = {
       initialState: null,
     };
+    this.stateRef = React.createRef();
+    this.stateRef.current = {
+      state: null,
+      lastChatIndex: 1,
+      lastRHPIndex: -1,
+      isRHPOnTopOfStack: false,
```

Comment:
Aaahh, yes. I agree none of this is needed if we just store the `state` because we can always derive all this info. We can have helper functions to help us with that.

Reply:
Just to be clear, you are saying we shouldn't even store `state` or you are just telling about these properties  `lastRightHandPanelScreenIndex`, `isRightHandPanelOnTopOfStack`(because yeah these two can be definitely calculated if we have the latest state info).

Reply:
For the purposes of this example app - I'm not seeing what is achieved by having the state stored here at all. The state will always be accessible from inside the navigator.

Reply:
>these two can be definitely calculated if we have the latest state info

Why are they getting calculated outside the navigator?